STM32U3: add WWDG and IWDG support

kbidani opened the pull request:
Add WWDG & IWDG support for STM32U3. Enable WWDG & IWDG on Nucleo U385RG_Q.

kbidani committed:
boards: st: stm32u3: enable wwdg for stm32u3 boards

Enable WWDG for Nucleo U385RG_Q board.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kbidani committed:
dts: arm: st: u3: add wwdg and iwdg

Add WWDG snd IWDG support to STM32U3

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

etienne-lms reviewed (commented):
Could you move commit "boards: st: stm32u3: enable wwdg for stm32u3 boards"
after commit "dts: arm: st: u3: add wwdg" in your series?

Otherwise LGTM.

kbidani committed:
tests: drivers: wdt_basic_api: Add nucleo u385rg_q as allowed platform

Add nucleo u385rg_q as allowed platform for iwdg and wwdg tests
to verify functionality on the STM32U3-series.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kartben commented:
please rebase

kbidani committed:
samples: drivers: watchdog: Add nucleo u385rg_q as allowed platform

Add iwdg and wwdg test for Nucleo U385RG-Q board
in watchdog driver sample.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

mathieuchopstm commented on dts/arm/st/u3/stm32u3.dtsi:
```diff
@@ -282,6 +282,20 @@
+
+		wwdg: watchdog@40002c00 {
+			compatible = "st,stm32-window-watchdog";
+			reg = <0x40002c00 0x1000>;
```
Nit:
```suggestion
			reg = <0x40002c00 0x400>;
```
or, if we wanted to be pedantic
```suggestion
			reg = <0x40002c00 DT_SIZE_K(1)>;
```

(Non-blocking as this size is never used; `dtc` *might* complain about overlapping ranges though?)

kartben commented:
@kbidani please rebase

etienne-lms reviewed (commented):
Nitpicking on commit messages:

```patch
 tests: drivers: wdt_basic_api: Add nucleo u385rg_q as allowed platform
 
-Add nucleo u385rg_q as allowed platform for wwdg tests
+Add nucleo u385rg_q as allowed platform for iwdg and wwdg tests
+in wdt_basic_api test
 to verify functionality on the STM32U3-series.
 
 Signed-off-by: ...
```
and 
```patch
 samples: drivers: watchdog: Add nucleo u385rg_q as allowed platform

-Add iwdg test for Nucleo U385RG-Q board.
+Add iwdg and wwdg test for Nucleo U385RG-Q board
+in watchdog driver sample.
```

Note: I'm not strongly requesting `+in wdt_basic_api test` and `in watchdog driver sample` additions. Just makses the message more consistent. Ignore it if your prefer. But at least state in both commits that both wwdg and iwdg are tested.
